pkg/chains/formats/slsa/internal/material/material.go

```diff
@@ -60,22 +60,29 @@ func AddImageIDToMaterials(imageID string, mats *[]common.ProvenanceMaterial) er
 	m := common.ProvenanceMaterial{
 		Digest: common.DigestSet{},
 	}
+	uriDigest, err := ExtractUriDigestFromImageID(imageID)
+	if err != nil {
+		return err
+	}
+	m.URI = uriDigest["uri"]
+	m.Digest[uriDigest["digestAlgorithm"]] = uriDigest["digestValue"]
+	*mats = append(*mats, m)
+	return nil
+}
+
+// ExtractUriDigestFromImageID extracts uri and digest from an imageID with format <uri>@sha256:<digest>
+func ExtractUriDigestFromImageID(imageID string) (map[string]string, error) {
 	uriDigest := strings.Split(imageID, uriSeparator)
 	if len(uriDigest) == 2 {
 		digest := strings.Split(uriDigest[1], digestSeparator)
 		if len(digest) == 2 {
-			// no point in partially populating the material
-			// do it if both conditions are valid.
-			m.URI = uriDigest[0]
-			m.Digest[digest[0]] = digest[1]
-			*mats = append(*mats, m)
+			return map[string]string{"uri": uriDigest[0], "digestAlgorithm": digest[0], "digestValue": digest[1]}, nil
```

Comment:
The three strings `uri`, `digestAlgorithm` and `digestValue` seem to be used at multiple places, can we declare them as constant?